Fixed issue with saving for multilingual rich text box

normangoh opened the pull request:
Voyager 1.2.3 have an issue with saving/editing multilingual rich text box.

The root cause of this issue is due to the position of DOM element for rich text box inputs. This changes will fix the issue by first targeting the parents of the "_il8n" element and then search for the input field with the class name "form-control".

normangoh committed:
Fixed issue with saving for multilingual rich text box

Voyager 1.2.3 have an issue with saving/editing multilingual rich text box.

The root cause of this issue is due to the position of DOM element for rich text box inputs. This changes will fix the issue by first targeting the parents of the "_il8n" element and then search for the input field with the class name "form-control".

jannescb reviewed (dismissed):
Tested and working on the following system:

Laravel 5.8.11
Voyager 1.2.3
PHP 7.2
MySQL 5.7.24

normangoh committed:
Update fixes.

Previous changes fixes issue with saving for multilingual rich text box but causes the menu item to be broken.

The current changes will detect if translatable.refresh() is calling from menu builder page and find the appropriate ".form-control" DOM element.

normangoh commented:
I noticed that my initial changes in the code caused menu builder to break and not showing the correct input values. Please refer to the new commit to resolve this issue.

g-trema reviewed (approved):
Tested with: 
Laravel: 5.8.4
Voyager: 1.2

emptynick commented:
Are you done here?

sarxan26 commented:
Tested with:
Laravel: 5.8.13
Voyager: 1.2.3

Not working, same problem

normangoh commented:
Those with issues after doing the changes that I pushed, you need to take note of the following.

1 - Make sure you run **npm run dev** at the path **/vendor/tcg/voyager**.
2 - Make sure to clear your browser cache.

NurdaKz commented:
![image](https://user-images.githubusercontent.com/50044259/56853359-58ce7d00-6948-11e9-9475-6c7897d87b86.png)
Hello! Lang select buttons dont work, after your fix. When I click on a language change, nothing happens.

Laravel: 5.8.13
Voyager: 1.2.3

normangoh commented:
> ![image](https://user-images.githubusercontent.com/50044259/56853359-58ce7d00-6948-11e9-9475-6c7897d87b86.png)
> Hello! Lang select buttons dont work, after your fix. When I click on a language change, nothing happens.
> 
> Laravel: 5.8.13
> Voyager: 1.2.3

I tried with 3 language set up and I am still able to toggle language without any issue. Can you try with 2 language set up to check if it is working? Your language control buttons seems to be all in an **active** state, I think something is wrong with there.

I wish to to check the following.
1 - Did you change anything else other than the files changes in the commit?
2 - Did you **npm run dev** at the **vendor/tcg/voyager** folder and then clear your browser cache?

NurdaKz commented:
> > ![image](https://user-images.githubusercontent.com/50044259/56853359-58ce7d00-6948-11e9-9475-6c7897d87b86.png)
> > Hello! Lang select buttons dont work, after your fix. When I click on a language change, nothing happens.
> > Laravel: 5.8.13
> > Voyager: 1.2.3
> 
> I tried with 3 language set up and I am still able to toggle language without any issue. Can you try with 2 language set up to check if it is working? Your language control buttons seems to be all in an **active** state, I think something is wrong with there.
> 
> I wish to to check the following.
> 1 - Did you change anything else other than the files changes in the commit?
> 2 - Did you **npm run dev** at the **vendor/tcg/voyager** folder and then clear your browser cache?

![image](https://user-images.githubusercontent.com/50044259/56875693-42b5df00-6a64-11e9-993a-d42256e8cdf6.png)

Only one active item. These styles are not correct.

1) I have not changed anything except what is in the commit
2) yes, I did it all.

I have RichTextBox works, no problem.

normangoh commented:
> > > ![image](https://user-images.githubusercontent.com/50044259/56853359-58ce7d00-6948-11e9-9475-6c7897d87b86.png)
> > > Hello! Lang select buttons dont work, after your fix. When I click on a language change, nothing happens.
> > > Laravel: 5.8.13
> > > Voyager: 1.2.3
> > 
> > 
> > I tried with 3 language set up and I am still able to toggle language without any issue. Can you try with 2 language set up to check if it is working? Your language control buttons seems to be all in an **active** state, I think something is wrong with there.
> > I wish to to check the following.
> > 1 - Did you change anything else other than the files changes in the commit?
> > 2 - Did you **npm run dev** at the **vendor/tcg/voyager** folder and then clear your browser cache?
> 
> ![image](https://user-images.githubusercontent.com/50044259/56875693-42b5df00-6a64-11e9-993a-d42256e8cdf6.png)
> 
> Only one active item. These styles are not correct.
> 
> 1. I have not changed anything except what is in the commit
> 2. yes, I did it all.
> 
> I have RichTextBox works, no problem.

Is there any errors in the browser console?

NurdaKz commented:
> Is there any errors in the browser console?

No. There are no errors in the console log

This my app.js 
https://codeshare.io/anQ4yX

ohmygod8 commented:
> > Is there any errors in the browser console?
> 
> No. There are no errors in the console log
> 
> This my app.js
> https://codeshare.io/anQ4yX

I cannot open the link you provided.

NurdaKz commented:
> > Is there any errors in the browser console?
> 
> No. There are no errors in the console log
> 
> This my app.js
> https://codeshare.io/anQ4yX

https://www.codepile.net/pile/qW37RbWO

skipper234 commented:
Hi, @ohmygod8 thanks for the code, sadly this doesn't worked for me instead I came up with some like:
```
var inpUsr = _inp.next(_this.settings.editing ? '.form-control' : '');
if (inpUsr.length == 0)
     inpUsr = $(_inp).parent().find(_this.settings.editing ? '.form-control' : '');
```

marijnz0r commented:
@ohmygod8 @skipper234 
Are you able to find the best solution for now and update the pull request?
I would appreciate it!

marijnz0r reviewed (approved):
This fix works for me!

mustafaabduljalil commented:
@marijnz0r this fix does not work for me  ????

emptynick commented:
Closing in favor of https://github.com/the-control-group/voyager/pull/4157
#4157 works perfectly fine for me. If it doesn't for you guys please let me know.
